apps/product/src/main/java/com/ecommerce/product/controller/CategoryController.java

```diff
@@ -0,0 +1,38 @@
+package com.ecommerce.product.controller;
+import com.ecommerce.product.model.Category;
+import com.ecommerce.product.service.CategoryService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/category")
+@RequiredArgsConstructor
+@Validated
+public class CategoryController {
+    private final CategoryService categoryService;
+
+    @GetMapping("/all")
+    public ResponseEntity<List<Category>> getAllCategories() {
+        return ResponseEntity.ok().body(categoryService.getAllCategories());
```

Comment:
It's not good to always send the `ok()` in the response. It's better to let the services decide what should be sent back to client, and controller only need to forward it. You also need to implement a [global error handling](https://www.baeldung.com/exception-handling-for-rest-with-spring) to create rest response base on exception case.

Reply:
Add global exception handler using @ControllerAdvice annotation in 06f6603f39848258c2d67889a97210d168409203